### PR TITLE
Use Middleware to intercept login page

### PR DESCRIPTION
**Pull request by Altahrim**

### CODE REVIEW

File: lib/Middleware/CanSeeLoginPageMiddleware.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ * Checks whether the "sharing check" is enabled
+ *
+ * @package OCA\LimitLoginToIp\Middleware
+ * @psalm-api
+ */
+class CanSeeLoginPageMiddleware extends Middleware {
+	private const LOGIN_METHOD_NAMES = [
+		'tryLogin',
+		'confirmPassword',
+		'showLoginForm',
```
Comment (nickvergessen) [inline lib/Middleware/CanSeeLoginPageMiddleware.php:44]: But yourself as a code owner on that file to be aware of changes? :D

File: lib/Exception/ForbiddenLoginException.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ */
+
+namespace OCA\LimitLoginToIp\Exception;
+
+use Exception;
+
+final class ForbiddenLoginException extends Exception {
+	public function __construct() {
+		parent::__construct('Login denied', 401);
```
Comment (nickvergessen) [inline lib/Exception/ForbiddenLoginException.php:31]: 403?

**Review by nickvergessen — APPROVED** [review]
LGTM but didn't actually execute.

File: lib/Middleware/CanSeeLoginPageMiddleware.php (hunk 1 above)
Comment (Altahrim) [reply lib/Middleware/CanSeeLoginPageMiddleware.php:44]: Good idea!

**Pull request merged**